testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/maven/metadata/MavenMergeMetadataIT.java

```diff
@@ -0,0 +1,204 @@
+
+  private Repository proxyRepository;
+
+  private Repository mavenGroup;
+
+  @Before
+  public void prepare() throws Exception {
+    upstream = Server.withPort(0)
+        .serve("/" + METADATA_PATH).withBehaviours(
+            new GeneratorBehaviour(new MavenMetadataGenerator(defaultMetadata("1.0-proxy")))
+        )
+        .start();
+
+    hostedRepository = repositoryManager.create(
+        hostedConfig("hosted", VersionPolicy.RELEASE)
+    );
+    proxyRepository = repositoryManager.create(
+        proxyConfig("proxy", "http://localhost:" + upstream.getPort() + "/", VersionPolicy.RELEASE)
+    );
+
+    deployMetadataWithVersionTo("1.0-hosted", hostedRepository);
+    mavenGroup = repositoryManager.create(
+        groupConfig(
+            "mavenGroup",
+            hostedRepository.getName(),
+            proxyRepository.getName()
+        )
+    );
+  }
+
+  @After
+  public void cleanup() throws Exception {
+    repositoryManager.delete(mavenGroup.getName());
+    repositoryManager.delete(proxyRepository.getName());
+    repositoryManager.delete(hostedRepository.getName());
+    upstream.stop();
+  }
+
+  private Metadata defaultMetadata() {
+    Metadata metadata = new Metadata();
+    metadata.setModelVersion("1.1.0");
+    metadata.setGroupId(GROUP_ID);
+    metadata.setArtifactId(ARTIFACT_ID);
+    metadata.setVersioning(new Versioning());
+    return metadata;
+  }
+
+  private Metadata defaultMetadata(String version) {
+    Metadata metadata = defaultMetadata();
+    metadata.getVersioning().getVersions().add(version);
+    return metadata;
+  }
+
+  private void deployMetadataWithVersionTo(String version, Repository repository) throws Exception {
+    Metadata metadata = get(repository);
+    if (metadata == null) {
+      metadata = defaultMetadata(version);
+    }
+    else if (!metadata.getVersioning().getVersions().contains(version)) {
+      metadata.getVersioning().addVersion(version);
+    }
+    StringWriter sw = new StringWriter();
+    writer.write(sw, metadata);
+    final Maven2Client client = createAdminMaven2Client(repository.getName());
+    HttpResponse response = client.put(METADATA_PATH, new StringEntity(sw.toString(), ContentType.TEXT_XML));
+    assertThat(response.getStatusLine().getStatusCode(), equalTo(201));
+  }
+
+  private Metadata get(final Repository repository) throws Exception {
+    final Maven2Client client = createAdminMaven2Client(repository.getName());
+    HttpResponse response = client.get(METADATA_PATH);
+    if (response.getStatusLine().getStatusCode() == 200) {
+      Metadata metadata = reader.read(response.getEntity().getContent());
+      EntityUtils.consume(response.getEntity());
+      return metadata;
+    }
+    else if (response.getStatusLine().getStatusCode() == 404) {
+      return defaultMetadata();
+    }
+    else {
+      // this should fail the test too
+      throw new IllegalStateException("Unexpected response: " + response.getStatusLine());
```

Comment:
within the context of the test it's probably better to use fail() than throw an exception